folio-spring-i18n/src/main/java/org/folio/spring/model/TranslationFile.java

```diff
@@ -0,0 +1,137 @@
+package org.folio.spring.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.Value;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.core.io.Resource;
+
+/**
+ * A class to represent a file that contains translation information.  This file will have a name
+ * identifying its associated locale with its contents being an ICU dictionary.
+ *
+ * <p>This class also contains several utility methods for parsing filenames and aggregating
+ * {@code TranslationFile} objects.</p>
+ */
+@Value
+@Log4j2
+public class TranslationFile {
+
+  /**
+   * The String denoting that a language or country is unknown from {@link getParts}.
+   */
+  public static final String UNKNOWN_PART = "*";
+
+  private static final int MAX_FILENAME_PARTS = 2;
+  private static final String JSON_FILE_SUFFIX = ".json";
+
+  /**
+   * The resources backing the translation files.
+   */
+  protected List<Resource> resources;
+
+  /**
+   * Get the map of patterns from this file.
+   *
+   * @return the map of patterns -&gt; ICU format strings
+   */
+  public Map<String, String> getMap() {
+    Map<String, String> map = new HashMap<>();
```

Comment:
Could we rename this? Even something simple like `out` or `ret` or something would make reading the code a bit easier